addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3242,6 +3242,12 @@ const SnippetOptionWidget = Widget.extend({
 
         const styles = window.getComputedStyle(this.$target[0]);
 
+        // Case where cssProperty is a CSS variable.
+        if (params.cssProperty.startsWith('--')) {
+            this.$target[0].style.setProperty(params.cssProperty, widgetValue);
+            return;
+        }
```

Comment:
My bad, I think I looked too fast into it sorry... The problem comes in fact from `styles[cssProp]` in `applyCSS`: we can't get a CSS variable by doing this. Instead, we have to do `styles.getPropertyValue(cssProp)` which works both with normal properties and CSS variables.

Reply:
```suggestion
        const propertyValue = styles.getPropertyValue(cssProp);
            // This condition requires extraClass to NOT be set.
            if (!weUtils.areCssValuesEqual(propertyValue, cssValue, cssProp, this.$target[0])) {
                // Property must be set => extraClass will be enabled.
                if (params.extraClass) {
                    // The extraClass is temporarily removed during selectStyle
                    // because it is enabled only if the element style is set
                    // by the option. (E.g. add the bootstrap border class only
                    // if there is a border width.) Unfortunately the
                    // extraClass might specify default !important properties,
                    // therefore determining whether !important is needed
                    // requires the class to be applied.
                    this.$target[0].classList.add(params.extraClass);
                    // Set inline style only if different from value defined
                    // with extraClass.
                    if (!weUtils.areCssValuesEqual(propertyValue, cssValue, cssProp, this.$target[0])) {
                        this.$target[0].style.setProperty(cssProp, cssValue);
                    }
                } else {
                    // Inline style required.
                    this.$target[0].style.setProperty(cssProp, cssValue);
                }
                // If change had no effect then make it important.
                // This condition requires extraClass to be set.
                if (!weUtils.areCssValuesEqual(propertyValue, cssValue, cssProp, this.$target[0])) {
                    this.$target[0].style.setProperty(cssProp, cssValue, 'important');
                }
                if (params.extraClass) {
                    this.$target[0].classList.remove(params.extraClass);
                }
                return true;
            }
            return false;
        }
```

I added the `propertyValue` variable and replaced the styles[cssProp] by that. Is it alright ?

Reply:
same in ComputeWidgetState: `styles[cssProp].trim()` => `styles.getPropertyValue(cssProp).trim()`

Reply:
> I added the propertyValue variable and replaced the styles[cssProp] by that (in applyCSS). Is it alright ?

It seems alright, you tell me :wink: You don't need to send diff in the answer (I can't read it anyway, it will be easier in the final diff), you can just hit the "resolve conversation" button if you adapted according to my comment in your local branch and you can push-force the branch at the end of the day as always (btw try to not push-force with a rebase and code changes at the same time, if you rebase make it a dedicated push-force, that will help me view the changes you made) :+1:

In the end, try to answer only if I asked a question or if you have a question yourself (so not when it is just about saying you agreed and made the diff) otherwise I will receive a mail for every comment I put in reviews and I have enough mails as it is, ah ah :smile: Of course, if you have questions, ping me and when you adapted everything, ping me too :wink: